AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,448 @@
+            break
+    return found_node
+
+def get_node_palette_node_tree_qt_object (self):
+    """
+    function for retrieving the tree view qt object for the node palette
+
+    params self: the script calling this function
+
+    returns: a tree view qt object
+    """
+    node_palette_widget = self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+    node_palette_node_tree = node_palette_widget.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+    return node_palette_node_tree
+
+
+def get_node_palette_category_qt_object(self, category_name):
+    """
+    function for retrieving the qt object for a node palette category
+
+    param self: the script calling this function
+    param category_name: string for the category label you are searching node palette for
+
+    returns: the qt object for the node palette category
+    """
+    node_palette_node_tree = get_node_palette_node_tree_qt_object(self)
+    node_palette_category = pyside_utils.find_child_by_pattern(node_palette_node_tree, category_name)
+    return node_palette_category
+
+def get_node_inspector_node_titles(self, sc_graph_node_inspector, sc_graph):
+    """
+    function for retrieving the node inspector's node titles from all nodes in a script canvas graph. function takes
+    a script canvas graph and node inspector qt widget.
+
+    param self: the script calling this function
+    param sc_graph_node_inspector: the sc graph node inspector qt widget
+    param sc_graph: the sc graph qt widget
+
+    returns: a list of node titles (i.e Print - Utilities/Debug). If there are duplicates of a node then the title
+    will include ( X Selected) in the string.
+    """
+    node_inspector_scroll_area = sc_graph_node_inspector.findChild(QtWidgets.QScrollArea, "")
+    # perform ctrl+a keystroke to highlight all nodes on the graph
+    QtTest.QTest.keyClick(sc_graph, "a", Qt.ControlModifier, WAIT_FRAMES_200)
+    node_inspector_backgrounds = node_inspector_scroll_area.findChildren(QtWidgets.QFrame, "Background")
+    titles = []
+    for background in node_inspector_backgrounds:
+        background_title = background.findChild(QtWidgets.QLabel, NODE_INSPECTOR_TITLE_KEY)
+        if background_title.text() != "":
+            titles.append(background_title.text())
+    return titles
+
+
+def get_sc_editor_node_inspector(sc_editor):
+    """
+    function for toggling the node inspector if it's not already turned on and returning the qt widget object
+
+    param sc_editor: the script canvas editor qt object
+
+    returns: the node inspector qt widget object
+
+    """
+    node_inspector_widget = sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT)
+    if sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT) is None:
+        action = pyside_utils.find_child_by_pattern(sc_editor, {"text": NODE_INSPECTOR_UI, "type": QtWidgets.QAction})
+        action.trigger()
+
+    return node_inspector_widget
+
+
+def create_script_event(self):
```

Comment:
Refactor to an asset editor uitls file or QtPy class.

Reply:
left a note on this to move it to the class once we implement it